namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+# MSON Namespace
+
+This document extends [Refract][]'s [JSON Namespace][] with elements necessary to
+build [MSON][] DOM.
+
+# Content
+
+This namespace defines following elements:
+
+1. General-use elements
+    1. [Id Element](#id-element-distinct-element)
+    1. [Distinct Element](#distinct-element-element)
+    1. [Select](#select-distinct-element)
+    1. [Option](#option-distinct-element)
+
+1. MSON DOM-specific elements
+    1. [MSON Element](#mson-element-distinct-element)
+    1. [Boolean Type](#boolean-type-jsonboolean-type)
+    1. [String Type](#string-type-jsonstring-type)
+    1. [Number Type](#number-type-jsonnumber-type)
+    1. [Array Type](#array-type-jsonarray-type)
+    1. [Object Type](#object-type-jsonobject-type)
+    1. [Property Type](#property-type-jsonproperty-type)
+    1. [Enum Type](#enum-type-mson-element)
+
+# Terminology
+
+This specification uses terminology from [MSON Specification][], in addition, bellow you can find the explanation of terms defined in this namespace.
+
+## Expanded Element
+
+MSON is built around the idea of defining recursive data structures. To provide abstraction, for convenience reasons and to not repeat ourselves these structures can be named (using an _identifier_) and reused. In MSON, the reusable data structures are called _Named Types_.
+
+Often, before an MSON DOM can be processed referenced _Named Types_ have to be resolved. Resolving references to _Named Types_ is tedious and error prone. As such parser can resolve references and produce a complete MSON DOM, that is a DOM that does not include unresolved references to other data structures. This is referred to as _reference expansion_ or simply _expansion_.
+
+In other words, an expanded element is such an element that does not contain _Identifier_ (defined bellow) to any other elements but those defined in JSON or MSON namespaces.
```

Comment:
I do propose the "ref" attribute (element) to take care of this. I actually think it might work. For example when you are rendering a documentation from such a DOM you may stop at first element with "ref" and display name (id) from that ref, knowing that it will be defined somewhere else...